src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -820,34 +847,53 @@ private static LevelsClause ParseLevels(long? levelsOption, IEdmType sourceType,
         }
 
         private static MetadataBinder BuildNewMetadataBinder(ODataUriParserConfiguration config,
+            IEdmNavigationSource resourcePathNavigationSource,
             IEdmNavigationSource targetNavigationSource,
             IEdmTypeReference elementType,
             HashSet<EndPathToken> generatedProperties = null,
             bool collapsed = false)
         {
-            BindingState state = CreateBindingState(config, targetNavigationSource, elementType, generatedProperties, collapsed);
+            BindingState state = CreateBindingState(config, resourcePathNavigationSource, targetNavigationSource, elementType, generatedProperties, collapsed);
             return new MetadataBinder(state);
         }
 
         private static BindingState CreateBindingState(ODataUriParserConfiguration config,
-            IEdmNavigationSource targetNavigationSource, IEdmTypeReference elementType,
-            HashSet<EndPathToken> generatedProperties = null, bool collapsed = false)
+            IEdmNavigationSource resourcePathNavigationSource, IEdmNavigationSource targetNavigationSource,
+            IEdmTypeReference elementType, HashSet<EndPathToken> generatedProperties = null,
+            bool collapsed = false)
         {
             if (targetNavigationSource == null && elementType == null)
             {
                 return null;
             }
 
+            // For example if we have https://url/Books?$expand=Authors($filter=Name eq $it/Name)
+            // $filter=Name will reference Authors(the expanded entity).
+            // $it/Name will reference Books(the resource identified by the path).
+            // The BindingState ImplicitRangeVariable property will store the $it that references the expanded/selected item (The Implicit Range Variable).

```

Comment:
I have to admit, I'm not sure I understand the two statements on Line 873 and 874...